apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -56,9 +56,13 @@ where
             if ErrorCodes::from_u32(tx.result.code).unwrap()
                 == ErrorCodes::InvalidSig
             {
-                let wrapper =
-                    WrapperTx::try_from(&Tx::try_from(tx.tx.as_ref()).unwrap())
-                        .unwrap();
+                let wrapper = if let Ok(TxType::Wrapper(wrapper)) =
+                    TxType::try_from(Tx::try_from(tx.tx.as_ref()).unwrap())
```

Comment:
I'm not sure if this unwrap is safe, in `shell/process_proposal`, it looks like there are many cases where `TxResult` is converted to `Response::ProcessProposal` with default (empty) tx bytes

Reply:
The orignal tx is saved in the shim. FinalizeBlock sees the exact same tx as ProcessProposal

Reply:
See abcipp_shim.rs lines 87-94